### PR TITLE
Make examples conform to best practices guidance

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes
As I somehow accidentally merged this, I will have to reopen. Looking forward to a final once over to make sure everything works before we merge.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/pipelines/run_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        PipelineConfigurationError: If the module does not have an attribute
+            with the given name.
+    """
+    try:
+        return getattr(module, attribute_name)
+    except AttributeError:
+        raise PipelineConfigurationError(
+            f"Unable to load '{attribute_name}' from"
+            f" file '{module.__file__}'"
+        ) from None
```
Comment (fa9r) [inline src/zenml/pipelines/run_pipeline.py:222]: what does `from None` do here? Can that be removed?

File: src/zenml/pipelines/run_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            elif isinstance(materializers_config, dict):
+                materializers = {
+                    output_name: _load_class_from_module(
+                        module, source, str(zenml_root)
+                    )
+                    for output_name, source in materializers_config.items()
+                }
+            else:
+                raise PipelineConfigurationError(
+                    f"Only `str` and `dict` values are allowed for "
```
Comment (fa9r) [inline src/zenml/pipelines/run_pipeline.py:130]: Error above says `str` is also not allowed now. Maybe we can combine those two errors into one that simply checks whether the materializer is a dict?

File: examples/feast_feature_store/feast_feature_repo/example.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 # production, you can use your favorite data warehouse, such as BigQuery. See Feast documentation
 # for more info.
 driver_hourly_stats = FileSource(
-    path="./data/driver_stats.parquet",
```
Comment (fa9r) [inline examples/feast_feature_store/feast_feature_repo/example.py:10]: What is this feast_feature_repo/ dir used for? Seems like it's not used in the example pipeline/steps/run.py code anywhere.

File: examples/feast_feature_store/config.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+name: feast_pipeline
```
Comment (fa9r) [inline examples/feast_feature_store/config.yaml:1]: For other integrations you mentioned to run via `-c config.yaml`, why not for feast?

File: examples/kubeflow_pipelines_orchestration/config.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+name: mnist_pipeline
```
Comment (fa9r) [inline examples/kubeflow_pipelines_orchestration/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not for Kubeflow?

File: examples/quickstart/inference_config.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+name: inference_pipeline
```
Comment (fa9r) [inline examples/quickstart/inference_config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not for quickstart?

File: examples/seldon_deployment/continuous_deployment_config_sklearn.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+name: continuous_deployment_pipeline
```
Comment (fa9r) [inline examples/seldon_deployment/continuous_deployment_config_sklearn.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not for Seldon?

File: examples/slack_alert/config.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+name: post_pipeline
```
Comment (fa9r) [inline examples/slack_alert/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not for Slack?

File: examples/step_operator_remote_training/config.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+name: step_operator_pipeline
```
Comment (fa9r) [inline examples/step_operator_remote_training/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not for this one?

File: examples/vertex_ai_orchestration/config.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+name: vertex_example_pipeline
```
Comment (fa9r) [inline examples/vertex_ai_orchestration/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not here?

File: examples/wandb_tracking/config.yaml [hunk 11]
```diff
@@ -0,0 +1,22 @@
+name: wandb_example_pipeline
```
Comment (fa9r) [inline examples/wandb_tracking/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not here?

File: examples/whylogs_data_profiling/config.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+name: data_profiling_pipeline
```
Comment (fa9r) [inline examples/whylogs_data_profiling/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not here?

File: examples/xgboost/config.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
+name: xgboost_pipeline
```
Comment (fa9r) [inline examples/xgboost/config.yaml:1]: for other examples you mentioned in the README the option to run with `-c config`, why not here?

**Review by fa9r — APPROVED** [review]
On a high level LGTM. Main questions I had:
- why is `-c config` used to run pipelines in some examples but not others
- what is this subfolder in Feast used for?
- Minor code style suggestions on the src/ changes

File: src/zenml/pipelines/run_pipeline.py (hunk 2 above)
Comment (AlexejPenner) [reply src/zenml/pipelines/run_pipeline.py:130]: For now I would leave it as two separate messages with the correction of removing str from the second message. Reason being, that if its a str (as it used to be) we can give some additional issue of how to easily change

File: examples/feast_feature_store/feast_feature_repo/example.py (hunk 3 above)
Comment (AlexejPenner) [reply examples/feast_feature_store/feast_feature_repo/example.py:10]: @strickvl I am not too sure here, any idea

**Comment by fa9r** [comment]
Sounds good to me, also looking forward to having this merged, I already approved anyways :D

File: examples/feast_feature_store/feast_feature_repo/example.py (hunk 3 above)
Comment (strickvl) [reply examples/feast_feature_store/feast_feature_repo/example.py:10]: It is used. There is a manual step that needs to be run from within that directory. It's needed :)

File: src/zenml/utils/source_utils.py [hunk 14]
```diff
@@ -523,47 +525,52 @@ def load_source_path_class(
...
+    # module_name = os.path.splitext(os.path.basename(file_path))[0]
 
     # In case the module is already fully or partially imported and the module
     #  path is something like materializer.materializer the full path needs to
     #  be checked for in the sys.modules to avoid getting an empty namespace
     #  module
-    full_module_path = os.path.splitext(
-        os.path.relpath(file_path, os.getcwd())
-    )[0].replace(os.path.sep, ".")
+    module_name = os.path.splitext(os.path.relpath(module_path, os.getcwd()))[
```
[the remark this answers is not in the document]
Comment (AlexejPenner) [reply src/zenml/utils/source_utils.py:555]: Good catch,  never noticed this as the working dir is usually also the root (at least in te bp)

**Pull request merged**